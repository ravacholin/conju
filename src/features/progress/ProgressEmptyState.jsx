import React from 'react'

export default function ProgressEmptyState({
  onSync,
  syncEnabled,
  syncing,
  onNavigateToDrill,
  onNavigateHome,
  systemReady,
  onRefresh
}) {
  const handleSync = React.useCallback(() => {
    if (!syncEnabled || !onSync) return
    onSync()
  }, [onSync, syncEnabled])

  const handleStartPractice = React.useCallback(() => {
    if (onNavigateToDrill) {
      onNavigateToDrill()
    }
  }, [onNavigateToDrill])

  const handleGoHome = React.useCallback(() => {
    if (onNavigateHome) {
      onNavigateHome()
    }
  }, [onNavigateHome])

  const handleRefresh = React.useCallback(() => {
    if (onRefresh) {
      onRefresh()
    }
  }, [onRefresh])

  return (
    <div className="progress-empty-state" data-testid="progress-empty-state">
      <img
        className="empty-state-illustration"
        src="/icons/lightbulb.png"
        alt="Sin datos de progreso"
      />

      <div className="empty-state-copy">
        <h2 className="empty-state-title">
          {systemReady ? 'Aún no tenemos datos de tu progreso' : 'Estamos preparando tu tablero'}
        </h2>
        <p className="empty-state-description">
          Sin sesiones sincronizadas no podemos generar estadísticas. Sincronizá tus avances o comenzá una práctica guiada
          para que el tablero cobre vida.
        </p>
      </div>

      <div className="empty-state-actions">
        <button
          type="button"
          className="empty-action primary"
          onClick={handleStartPractice}
          disabled={!onNavigateToDrill}
        >
          Iniciar práctica
        </button>

        <button
          type="button"
          className="empty-action secondary"
          onClick={handleSync}
          disabled={!syncEnabled || syncing || !onSync}
        >
          {syncing ? 'Sincronizando…' : 'Sincronizar progreso'}
        </button>

        <button type="button" className="empty-action ghost" onClick={handleRefresh} disabled={!onRefresh}>
          Reintentar carga
        </button>

        <button type="button" className="empty-action ghost" onClick={handleGoHome} disabled={!onNavigateHome}>
          Volver al inicio
        </button>
      </div>
    </div>
  )
}
