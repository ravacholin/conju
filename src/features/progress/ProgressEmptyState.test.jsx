import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'

import ProgressEmptyState from './ProgressEmptyState.jsx'

describe('ProgressEmptyState', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <ProgressEmptyState
        syncEnabled
        onSync={() => {}}
        onNavigateToDrill={() => {}}
        onNavigateHome={() => {}}
        onRefresh={() => {}}
        systemReady
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('disables sync button when syncing or sync disabled', () => {
    const onSync = vi.fn()
    const { rerender } = render(
      <ProgressEmptyState
        syncEnabled={false}
        syncing={false}
        onSync={onSync}
        onNavigateToDrill={() => {}}
        systemReady
      />
    )

    const syncButton = screen.getByRole('button', { name: /sincronizar progreso/i })
    expect(syncButton).toBeDisabled()

    rerender(
      <ProgressEmptyState
        syncEnabled
        syncing
        onSync={onSync}
        onNavigateToDrill={() => {}}
        systemReady
      />
    )

    expect(screen.getByRole('button', { name: /sincronizando/i })).toBeDisabled()
  })

  it('invokes handlers when actions are clicked', () => {
    const onSync = vi.fn()
    const onPractice = vi.fn()
    const onRefresh = vi.fn()
    const onHome = vi.fn()

    render(
      <ProgressEmptyState
        syncEnabled
        onSync={onSync}
        onNavigateToDrill={onPractice}
        onRefresh={onRefresh}
        onNavigateHome={onHome}
        systemReady
      />
    )

    fireEvent.click(screen.getByRole('button', { name: /iniciar pr√°ctica/i }))
    fireEvent.click(screen.getByRole('button', { name: /sincronizar progreso/i }))
    fireEvent.click(screen.getByRole('button', { name: /reintentar carga/i }))
    fireEvent.click(screen.getByRole('button', { name: /volver al inicio/i }))

    expect(onPractice).toHaveBeenCalledTimes(1)
    expect(onSync).toHaveBeenCalledTimes(1)
    expect(onRefresh).toHaveBeenCalledTimes(1)
    expect(onHome).toHaveBeenCalledTimes(1)
  })
})
