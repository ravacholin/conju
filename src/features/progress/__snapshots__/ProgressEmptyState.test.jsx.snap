// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProgressEmptyState > matches snapshot 1`] = `
<div
  class="progress-empty-state"
  data-testid="progress-empty-state"
>
  <img
    alt="Sin datos de progreso"
    class="empty-state-illustration"
    src="/icons/lightbulb.png"
  />
  <div
    class="empty-state-copy"
  >
    <h2
      class="empty-state-title"
    >
      Aún no tenemos datos de tu progreso
    </h2>
    <p
      class="empty-state-description"
    >
      Sin sesiones sincronizadas no podemos generar estadísticas. Sincronizá tus avances o comenzá una práctica guiada para que el tablero cobre vida.
    </p>
  </div>
  <div
    class="empty-state-actions"
  >
    <button
      class="empty-action primary"
      type="button"
    >
      Iniciar práctica
    </button>
    <button
      class="empty-action secondary"
      type="button"
    >
      Sincronizar progreso
    </button>
    <button
      class="empty-action ghost"
      type="button"
    >
      Reintentar carga
    </button>
    <button
      class="empty-action ghost"
      type="button"
    >
      Volver al inicio
    </button>
  </div>
</div>
`;
